src/platform/nrfconnect/wifi/WiFiManager.cpp

```diff
@@ -205,6 +205,7 @@ CHIP_ERROR WiFiManager::Scan(const ByteSpan & ssid, ScanResultCallback resultCal
     mCachedWiFiState    = mWiFiState;
     mWiFiState          = WIFI_STATE_SCANNING;
     mSsidFound          = false;
+    mRecoveryStarted    = true;
```

Comment:
Seems weird that we start recovery in the scan request.

Reply:
Why? I mean, maybe the field name is not the best, as it's not like we start recovery here, but rather enable possibility of starting it. Would naming it `mRecoveryEnabled` be better for you?

Reply:
Sounds better. Or, following Matter FailSafe fashion - `mRecoveryArmed` :). It's a nit, we can leave it as it is unless you are introducing more changes anyway.

Reply:
> Or, following Matter FailSafe fashion - mRecoveryArmed :).

I like it! Done.